drv/i2c-devices/src/max31790.rs

```diff
@@ -311,7 +320,18 @@ impl Max31790 {
         const NP: u32 = 2;
         const FREQ: u32 = 8192;
 
-        if count == TACH_POR_VALUE {
+        if count == 0 {
+            //
+            // We don't really expect this:  generally, if a fan is off (or is
+            // otherwise emiting non-detectable tach input pulses), the
+            // controller will report the power-on-reset value for the tach
+            // count, not 0.  So if we see a zero count, we will assume that
+            // this is an error rather than a 0 RPM reading, and record it to
+            // a (small) ring buffer and return accordingly.
+            //
+            ringbuf_entry!(Trace::ZeroTach(fan));
```

Comment:
This ringbuf isn't strictly necessary, because the error will bubble up and be logged in the thermal ringbuf as `FanReadFailed`, but it's small enough that I'm not concerned.

Reply:
Agreed -- part of the rationale there was to get these failures into a ringbuf that won't roll over so we can essentially know if this happened at any time in the past.